### PR TITLE
Get out of dependency 'heck'

### PR DESCRIPTION
**Pull request by kienstra**
## Before
Every time I updated npm packages, that caused [errors](https://app.circleci.com/pipelines/github/studiopress/genesis-custom-blocks/1611/workflows/83d6e92b-235e-439a-a5d5-53ef1bbf0295/jobs/13040) with Jest unit tests or `npm run build`. 

And `composer install` failed on PHP `8.2`, as the packages weren't compatible with `8.2`.

## After
The npm package versions don't matter much anymore. 

And it'll be easier to update package versions, without a Jest or PHPUnit error.

## Changes
* Remove React component unit tests. Testing the markup layer was a bad idea (that I had). It meant that every time we updated the npm packages, Jest tests could fail, like from `React` versions being out of sync.
* Remove `--webpack-no-externals`. This should make npm package updates less necessary. The block editor will alias them to whatever packages it's using.
* Update [@wordpress/*](https://www.npmjs.com/package/@wordpress/scripts) dependencies
* Make unit and e2e tests pass

## Testing instructions
Not needed

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,8 +4,6 @@
     "plugin:@wordpress/eslint-plugin/recommended-with-formatting",
     "plugin:import/recommended",
     "plugin:eslint-comments/recommended",
-    "plugin:jsx-a11y/recommended",
-    "plugin:jest-dom/recommended",
```
Comment (kienstra) [inline .eslintrc:8]: This was for Jest unit tests of React components, which I deleted

File: composer.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "homepage": "https://github.com/studiopress/genesis-custom-blocks",
     "license": "GPL-2.0",
     "require": {
-        "php": "^5.6 || ^7 || ^8"
+        "php": "^7 || ^8"
```
Comment (kienstra) [inline composer.json:8]: Core `6.3` [doesn't support](https://make.wordpress.org/core/2023/07/05/dropping-support-for-php-5/) PHP `5.6` anymore. 

That's a good excuse to remove our support, as it's really hard to make `composer install` work for PHP `5.6`-`8.2`.

File: tests/e2e/specs/all-fields.js [hunk 3]
```diff
@@ -23,7 +22,7 @@ const uploadMediaFile = async ( $context, fieldLabel, fileName ) => {
 	const $input = await page.$( inputSelector );
 
 	const testImagePath = path.join( __dirname, '..', 'assets', fileName );
-	const newFileName = uuid();
+	const newFileName = Math.floor( Math.random( 100000 ) * 100000 );
```
Comment (kienstra) [inline tests/e2e/specs/all-fields.js:25]: No need to import [uuid](https://www.npmjs.com/package/uuid), we can do this in 1 line of JS.

File: package.json [hunk 4]
```diff
@@ -98,20 +90,18 @@
     "react-ace": "9.4.0",
     "react-autosize-textarea": "7.1.0",
     "react-dom": "^17.0.1",
-    "resize-observer-polyfill": "1.5.1",
```
Comment (kienstra) [inline package.json:101]: Not needed, now that we're not unit testing React components.

File: composer.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
         "mockery/mockery": "^1",
         "phpcompatibility/phpcompatibility-wp": "^2.1.3",
         "squizlabs/php_codesniffer": "^3.5.3",
-        "wp-coding-standards/wpcs": "^2.2.0",
+        "wp-coding-standards/wpcs": "dev-develop",
```
Comment (kienstra) [inline composer.json]: `dev-develop` has a fix we need for PHP `8`

File: package.json [hunk 6]
```diff
@@ -98,20 +90,18 @@
...
     "sass": "1.49.8",
     "sass-loader": "9.0.3",
     "style-loader": "1.2.1",
     "tailwindcss": "2.0.2",
     "typescript": "4.5.5",
-    "uuid": "8.3.2",
```
Comment (kienstra) [inline package.json:107]: I added too many NPM packages. This one is [replaced with](https://github.com/studiopress/genesis-custom-blocks/pull/145/files#r1274062822) a 1-liner of vanilla JavaScript 🤣

File: package.json [hunk 7]
```diff
@@ -98,20 +90,18 @@
...
     "webpack": "5.69.1",
     "webpack-cli": "4.9.2"
   },
   "scripts": {
     "check-licenses": "wp-scripts check-licenses --prod --gpl2",
-    "build": "wp-scripts build --webpack-no-externals",
-    "dev": "wp-scripts start --webpack-no-externals",
```
Comment (kienstra) [inline package.json:114]: `--webpack-no-externals` made updating packages important.

Without it, the block editor will alias these packages to whatever it's using.

File: php/Admin/Admin.php [hunk 8]
```diff
@@ -78,7 +80,7 @@ public function init() {
 		genesis_custom_blocks()->register_component( $this->upgrade );
 
 		if ( defined( 'WP_LOAD_IMPORTERS' ) && WP_LOAD_IMPORTERS ) {
-			$this->import = new Import();
+			$this->import = new Import( $wp_filesystem );
```
Comment (kienstra) [inline php/Admin/Admin.php:86]: Dependency injection, so the PHPUnit test can inject a stub filesystem class.

File: tests/bootstrap.php [hunk 9]
```diff
@@ -5,15 +5,16 @@
  * @package PluginTest
  */
 
+define( 'WP_RUN_CORE_TESTS', true );
```
Comment (kienstra) [inline tests/bootstrap.php:8]: To prevent warnings in PHPUnit.

Copied from Gutenberg.

File: .circleci/config.yml [hunk 10]
```diff
@@ -130,7 +130,7 @@ workflows:
               only: /.*/
           matrix:
             parameters:
-              php-version: [ '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1' ]
+              php-version: [ '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2' ]
```
Comment (kienstra) [inline .circleci/config.yml:154]: Now, we can unit test on PHP `8.2`

**Comment by kienstra** [comment]
Hi @dreamwhisper,
Would you have some time to look at this?

I let GCB get into dependency 'heck' 😄

**Review by dreamwhisper — APPROVED** [review]

**Comment by kienstra** [comment]
Hi @dreamwhisper,
Thanks a lot for reviewing this!

**Pull request merged**